egui/src/widgets/slider.rs

```diff
@@ -346,15 +385,85 @@ impl<'a> Slider<'a> {
                 // stroke: ui.visuals().widgets.inactive.bg_stroke,
             });
 
+            let center = self.marker_center(position_1d, &rail_rect);
+
             ui.painter().add(epaint::CircleShape {
-                center: pos2(marker_center_x, rail_rect.center().y),
-                radius: handle_radius(rect) + visuals.expansion,
+                center,
+                radius: self.handle_radius(rect) + visuals.expansion,
                 fill: visuals.bg_fill,
                 stroke: visuals.fg_stroke,
             });
         }
     }
 
+    fn marker_center(&self, position_1d: f32, rail_rect: &Rect) -> Pos2 {
+        match self.orientation {
+            SliderOrientation::Horizontal => pos2(position_1d, rail_rect.center().y),
+            SliderOrientation::Vertical => pos2(rail_rect.center().x, position_1d),
+        }
+    }
+
+    fn pointer_position(&self, pointer_position_2d: Pos2) -> f32 {
+        match self.orientation {
+            SliderOrientation::Horizontal => pointer_position_2d.x,
+            SliderOrientation::Vertical => pointer_position_2d.y,
+        }
+    }
+
+    fn position_range(&self, rect: &Rect) -> RangeInclusive<f32> {
+        let handle_radius = self.handle_radius(rect);
+        match self.orientation {
+            SliderOrientation::Horizontal => {
+                (rect.left() + handle_radius)..=(rect.right() - handle_radius)
+            }
+            SliderOrientation::Vertical => {
+                (rect.top() + handle_radius)..=(rect.bottom() - handle_radius)
```

Comment:
I think the default for a vertical slider should be higher values higher up, and lower values lower down:

```suggestion
                (rect.bottom() - handle_radius)..=(rect.top() + handle_radius)
```